demux/utils/novaseq_samplesheet.py

```diff
@@ -0,0 +1,203 @@
+        )
+
+    def add_dummy_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """Add all dummy indexes to raw sample sheet. Dummy indexes are used to check for index
+        contamination"""
+        with open(f"{self.dummy_indexes_file}") as csv_file:
+            dummy_samples_csv = csv.reader(csv_file, delimiter=COMMA)
+            dummy_samples = [row for row in dummy_samples_csv]
+            added_dummy_samples = []
+            lanes = {sample["lane"] for sample in raw_samplesheet}
+
+            for lane in lanes:
+                sample_indexes = self.sample_indexes_in_lane(raw_samplesheet, lane)
+                for name, dummy_index in dummy_samples:
+                    if not self.is_dummy_sample_in_samplesheet(
+                        dummy_index, sample_indexes
+                    ):
+                        add_dummy_sample = self.add_dummy_sample(
+                            self.flowcell, dummy_index, lane, name
+                        )
+                        added_dummy_samples.append(add_dummy_sample)
+
+            self.raw_samplesheet.extend(added_dummy_samples)
+
+            return raw_samplesheet
+
+    def remove_unwanted_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """ Filter out indexes of unwanted length and single indexes """
+
+        raw_samplesheet = [
+            line for line in raw_samplesheet if self.is_dual_index(line["index"])
+        ]
+        return raw_samplesheet
+
+    def adapt_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
+        number  of index reads, and takes the reverse complement of index 2 in case of the new
+        novaseq software control version (1.7) in combination with the new reagent kit
+        (version 1.5)"""
+
+        reverse_complement = self.is_reverse_complement()
+
+        for line in raw_samplesheet:
+            index1, index2 = line["index"].split("-")
+            if self.pad and len(index1) == 8:
+                line["index"], line["index2"] = self.pad_and_rc_indexes(
+                    index1, index2, reverse_complement
+                )
+            elif len(index2) == 10:
+                line["index2"] = (
+                    self.reverse_complement(index2) if reverse_complement else index2
+                )
+            else:
+                line["index"], line["index2"] = index1, index2
+
+        return raw_samplesheet
+
+    def pad_and_rc_indexes(
+        self, index1: str, index2: str, reverse_complement: bool
+    ) -> tuple:
+        """ Pads and reverse complements indexes """
+
+        if self.runparameters.index_reads == 8:
+            index2 = self.reverse_complement(index2) if reverse_complement else index2
+        if self.runparameters.index_reads == 10:
+            index1 = index1 + "AT"
+            index2 = (
+                self.reverse_complement("AC" + index2)
```

Comment:
Is there a good named constant for "AC" to better understand why this is added? Like PAD_??